forcedotcom/salesforcedx-vscode-automation-tests
# sObjectsDefinitions.test.ts runs successfully on remote

### How to test:
(1) Go to the [End to End Tests Workflow](https://github.com/forcedotcom/salesforcedx-vscode/actions/workflows/e2e.yml) in salesforcedx-vscode repo
(2) Paste the PR's branch (daphne/debugApexTests-on-remote) in the second field of the Run workflow dropdown (Set the branch to use for automation tests)
(3) Choose sObjectsDefinitions.e2e.ts test from the dropdown to select the test to run.

AC:
Test suite should pass.

### Key changes made:

(1) Used notificationIsPresentWithTimeout() instead of attemptToFindNotification().  This way, the program searches for notifications within a set period of time instead of within a set number of tries.

(2) Used getVisibleChild() instead of findChildItem() for finding Customer__c.object-meta.xml and Product__c.object-meta.xml within their parent folders.  This is because findChildItem() doesn't work as expected on Windows - the parent folders of those metadata files don't recognize that they have child items even though those child items are visible in the error screenshot.

### Successful runs:
• [368](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5929048207)
• [369](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5929049665)
• [370](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/5929051554)

@W-13744003@
